src/common/use-local-storage.ts

```diff
@@ -17,10 +17,21 @@ export function useLocalStorage<T>(
     const value = localStorage.getItem(key);
     if (value !== null) {
       try {
-        const parsed = JSON.parse(value);
+        let parsed = JSON.parse(value);
+        // Ensure we get new top-level defaults.
+        parsed = {
+          ...defaultValue,
+          ...parsed,
+        };
+        // Remove any top-level keys that aren't present in defaults.
+        const unknownKeys = new Set(Object.keys(parsed));
+        Object.keys(defaultValue).forEach((k) => unknownKeys.delete(k));
+        unknownKeys.forEach((k) => delete parsed[k]);
+
```

Comment:
We should test this or perhaps swap it out for something that's solved this problem well (though it could easily be excessive code).

Reply:
Test added in 3a4eb4b655522fb796c8ef6a3d5891614708b592.